chrome/content_script.js

```diff
@@ -442,14 +452,18 @@ passwordalert.completePageInitializationIfReady_ = function() {
       // Delete any previously considered password in case this is a re-prompt
       // when an incorrect password is entered, such as a ServiceLoginAuth page.
       chrome.runtime.sendMessage({action: 'deletePossiblePassword'});
-      var loginForm = document.getElementById('gaia_loginform');
-      // The chooser is also a gaia_loginform, so verify we're on a password
-      // entry page.
+      var loginForm = document.querySelector('#view_container > form');
+      // The chooser is no longer a gaia_loginform, so verify we're on a password
+      // entry page by finding a form in a view_container.
       if (loginForm && document.getElementById('Email')) {
         loginForm.addEventListener(
             'submit', passwordalert.saveGaiaPassword_, true);
       } else if (document.getElementById('hiddenEmail') &&
                  document.getElementsByName('password')){
```

Comment:
(not in your PR, in the previous code)
It looks like the check for document.getElementsByName('password') does not do anything. Even if it's not found, it will return an empty array which will evaluate to true.

Reply:
Huh, good point. Didn't think of that. Do you suggest removing it or fixing the check to see if the element is actually in the array?